mitodl/bootcamp-ecommerce
# Acceptance/rejection letters

#### What are the relevant tickets?
Fixes #600 

#### What's this PR do?
When a user is approved by the reviewer for all submissions, or if the user is rejected for any submission, an email will be sent to the user informing them of the decision. The text for acceptance or rejection is configurable in wagtail via a template using rich text fields. The text is rendered with variables relevant to the application, and a copy is stored in `ApplicantLetter`. The user can view their letter via the `/letters/<uuid>/` view, though nothing links to this page yet.

#### How should this be manually tested?

 - Go to `/cms/` and create a Letter Template Page as a child to the `HomePage`. You should see default text for the acceptance and rejection fields. 
 - You should be able to add a syntax error in the Django template tag (ie, `{%` without a closing tag). Click save and you should get a validation error.
  - Change the name of one of the variables to include a typo and click save. You should get a validation error.
  - Click 'Preview'. You should see a page which contains both the applicant and rejection text, and has fake information about an applicant. It should make it obvious that the page is sample text with a banner at the top.
  - Mark up the text using the rich text options. For example add a link, and make other text bold.
  - Publish the page, then click the live link. You should see a 404 error.
  - Then, go to `/review/<submission_id>/` for a submission.
  - Reject a submission. You should have a rejection letter emailed to the applicant. A new `ApplicantLetter` should appear with its text.
  - Approve all submissions. You should have an acceptance letter emailed to the applicant. Check that the link you added and any formatting changes show up in the email. A new `ApplicantLetter` should also appear with its text.
  - Pick a `ApplicantLetter`. Go to `/letters/<uuid>/` with `uuid` being `ApplicantLetter.hash`, including the dashes. You should see the letter that was sent.
  - Check that the formatting you added is displayed correctly, and that the link you added works.

#### Screenshots
Email received (note images are missing since links go to local server)
![Screenshot from 2020-06-26 18-35-36](https://user-images.githubusercontent.com/863262/85906307-e4a59880-b7db-11ea-9c62-d069a3b43b87.png)


Preview in CMS
![Screenshot from 2020-06-23 17-04-10](https://user-images.githubusercontent.com/863262/85906249-b6c05400-b7db-11ea-9c3e-5c081b928793.png)


A validation error example
![Screenshot from 2020-06-24 13-04-19](https://user-images.githubusercontent.com/863262/85906236-ad36ec00-b7db-11ea-9c7e-5d75d2aef252.png)
